includes/abstracts/abstract-wc-product.php

```diff
@@ -1420,8 +1420,10 @@ public function is_visible() {
 			$visible = false;
 		}
 
-		if ( ( $parent_id = wp_get_post_parent_id( $this->get_id() ) ) && 'publish' !== get_post_status( $parent_id ) ) {
-			$visible = false;
+		if ( !empty( $this->get_parent_id() ) ) {
```

Comment:
This will cause a fatal error on PHP 5.2 and 5.3.

>  Fatal error: Can't use function return value in write context in

Reply:
Updated the if statement to be compatible with older versions of PHP.